internal/command/apps/restart.go

```diff
@@ -72,9 +72,14 @@ func runRestart(ctx context.Context) error {
 }
 
 func runNomadRestart(ctx context.Context, app *api.AppCompact) error {
-	command.PromptToMigrate(ctx)
 	client := client.FromContext(ctx).API()
 
+	appName := flag.FirstArg(ctx)
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		command.PromptToMigrate(ctx, app)
+	}
```

Comment:
Getting the app is unnecessary here, we already have one passed as an argument to the function :)
```suggestion
	command.PromptToMigrate(ctx, app)
```